### PR TITLE
Fix single file package dynamic library

### PR DESCRIPTION
**Pull request by andre2007**
This PR fixes the issue that main sourcefile was not added to sourceFiles while building dynamic link library from dub single file package.

### CODE REVIEW

File: test/issue1505-single-file-package-dynamic-library.sh [hunk 1]
```diff
@@ -0,0 +1,12 @@
+#!/usr/bin/env bash
+. $(dirname "${BASH_SOURCE[0]}")/common.sh
+cd ${CURR_DIR}
+rm -f libsingle-file-test-dynamic-library.so
+rm -f single-file-test-dynamic-library.dll
+
+${DUB} build --single issue1505-single-file-package-dynamic-library.d
+if [ ! -f libsingle-file-test-dynamic-library.so ] && [ ! -f single-file-test-dynamic-library.dll ]; then
```
Comment (PetarKirov) [inline test/issue1505-single-file-package-dynamic-library.sh:8]:
```suggestion
if [ ! -f libsingle-file-test-dynamic-library.{dll,so,dylib} ]; then
```

File: test/issue1505-single-file-package-dynamic-library.sh [hunk 2]
```diff
@@ -0,0 +1,12 @@
+#!/usr/bin/env bash
+. $(dirname "${BASH_SOURCE[0]}")/common.sh
+cd ${CURR_DIR}
+rm -f libsingle-file-test-dynamic-library.so
+rm -f single-file-test-dynamic-library.dll
```
Comment (PetarKirov) [inline test/issue1505-single-file-package-dynamic-library.sh:5]:
```suggestion
rm -f single-file-test-dynamic-library.{dll,so,dylib}
```

File: test/issue1505-single-file-package-dynamic-library.sh [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+
+${DUB} build --single issue1505-single-file-package-dynamic-library.d
+if [ ! -f libsingle-file-test-dynamic-library.so ] && [ ! -f single-file-test-dynamic-library.dll ]; then
+	die $LINENO 'Normal invocation did not produce a dynamic library in the current directory'
+fi
```
Comment (PetarKirov) [inline test/issue1505-single-file-package-dynamic-library.sh:10]: The test would be more robust if it checks whether an exported function defined by the single-file shared library yields the expected result when called from another single file executable.

**Review by PetarKirov — COMMENTED** [review]
A few comments, otherwise the change looks good:

File: test/issue1505-single-file-package-dynamic-library.sh (hunk 1 above)
Comment (andre2007) [reply test/issue1505-single-file-package-dynamic-library.sh:8]: The windows `dll` and the linux `so` have different names. The linux so starts with `lib`.

Yes, I was also wondering why the MacOS test pipeline were succeeding although the `dylib` was missing.

File: test/issue1505-single-file-package-dynamic-library.sh (hunk 3 above)
Comment (andre2007) [reply test/issue1505-single-file-package-dynamic-library.sh:10]: @PetarKirov 
My gut feeling is, this rather complex logic is not necessary in this case. When dub and dmd returns with return code 0 and dmd produced a dynamic library, it is safe to assume, the file is valid. A test whether dmd is producing valid dynamic libraries I would see in the test suite of dmd.

**Comment by andre2007** [comment]
@PetarKirov I added the `dylib` extension. As already written the filename on windows and posix are different (`lib` prefix on posix) therefore I cannot combine them into single statements.

File: test/issue1505-single-file-package-dynamic-library.sh (hunk 1 above)
Comment (PetarKirov) [reply test/issue1505-single-file-package-dynamic-library.sh]: > The linux so starts with lib

Ah yes, then this should do it:

```
{lib,}single-file-test-dynamic-library.{dll,so,dylib}
```

File: test/issue1505-single-file-package-dynamic-library.sh (hunk 3 above)
Comment (PetarKirov) [reply test/issue1505-single-file-package-dynamic-library.sh:10]: If we had a test helper script (also used for the rest of the tests) that did such checks my case would be easier to make. In the absence of such, I'm fine with leaving it like this.

**Review by PetarKirov — APPROVED** [review]

**Pull request merged**